src/models/m_wikiconcept.erl

```diff
@@ -290,6 +299,92 @@ filter_lang(#trans{ tr = Tr }, Context) ->
     #trans{ tr = Tr2 }.
 
 
+%% @doc Handle the merge of two keywords. Merge all connected concepts.
+-spec rsc_merge(WinnerId, LoserId, Context) -> ok when
+    WinnerId :: m_rsc:resource_id(),
+    LoserId :: m_rsc:resource_id(),
+    Context :: z:context().
+rsc_merge(WinnerId, LoserId, Context) ->
+    case z_db:q("
+        update wikiconcept
+        set keyword_id = $1
+        where keyword_id = $2",
+        [ WinnerId, LoserId ],
+        Context)
+    of
+        0 ->
+            ok;
+        Count when is_integer(Count) ->
+            publish_update(WinnerId, Context)
+    end.
+
+%% @doc Find the keyword best matching to the given concept.
+%% First tries to find a keyword in the ascendants of a concept,
+%% then tries to find it in the descendents.
+-spec find_keyword(ConceptID, Context) -> {ok, KeywordRscId} | {error, Reason} when
+    ConceptID :: binary() | string(),
+    Context :: z:context(),
+    KeywordRscId :: m_rsc:resource_id(),
+    Reason :: term().
+find_keyword(ConceptID, Context) ->
+    case get_concept(ConceptID, Context) of
+        {ok, #{
+            <<"keyword_id">> := RscId
+        }} when is_integer(RscId) ->
+            {ok, RscId};
+        {ok, #{
+            <<"wikidata_id">> := WikidataID,
+            <<"wikidata_ancestor_ids">> := Ancestors
+        }} ->
+            case find_nearest_ancestor(Ancestors, Context) of
+                {ok, RscId} ->
+                    {ok, RscId};
+                {error, enoent} ->
+                    find_nearest_descendent(WikidataID, Context)
+            end;
+        {error, _} = Error ->
+            Error
+    end.
+
+find_nearest_ancestor(undefined, _Context) ->
+    {error, enoent};
+find_nearest_ancestor([], _Context) ->
+    {error, enoent};
+find_nearest_ancestor(Ancestors, Context) ->
+    case z_db:q1("
+        select keyword_id
+        from wikiconcept
+        where keyword_id is not null
+          and wikidata_id in (select(unnest($1::character varying[])))
```

Comment:
Minor suggestion, but AFAIK this could be more easily expressed as:
```suggestion
          and wikidata_id = ANY($1)
```
or, if the type inference needs a hand, as:
```suggestion
          and wikidata_id = ANY($1::character varying[])
```

Reply:
Indeed - updating this.